src/core/packages/saved-objects/migration-server-internal/src/actions/read_with_pit.test.ts

```
@@ -7,25 +7,19 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import { catchRetryableEsClientErrors } from './catch_retryable_es_client_errors';
+import * as Either from 'fp-ts/lib/Either';
 import { errors as EsErrors } from '@elastic/elasticsearch';
 import { elasticsearchClientMock } from '@kbn/core-elasticsearch-client-server-mocks';
 import { readWithPit } from './read_with_pit';
-
-jest.mock('./catch_retryable_es_client_errors');
+import * as errorHandlers from './catch_retryable_es_client_errors';
 
 describe('readWithPit', () => {
   beforeEach(() => {
     jest.clearAllMocks();
   });
   it('calls esClient.search with the appropriate params', async () => {
     const client = elasticsearchClientMock.createInternalClient(
-      elasticsearchClientMock.createSuccessTransportRequestPromise({
-        hits: {
-          total: 0,
-          hits: [],
-        },
-      })
+      Promise.resolve({ hits: { hits: [] } })
```

Changed this because the previous version returned:

```
{
      body: { hits: { total: 0, hits: [] } },
      statusCode: 200,
      headers: { 'x-elastic-product': 'Elasticsearch' },
      warnings: [],
      meta: {}
}
```

And this was causing: 

```
    TypeError: Cannot read properties of undefined (reading 'total')

      87 |         const totalHits =
    > 88 |           typeof body.hits.total === 'number'
```

This error didn't show because the handler was mocked and it just discarded the error

---

That's a really crappy mock I did there, thanks for fixing!